netdeploy/remote/deployedNetwork.go

```diff
@@ -283,9 +338,424 @@ func (cfg DeployedNetwork) BuildNetworkFromTemplate(buildCfg BuildConfig, rootDi
+	accounts[src] = basics.MakeAccountData(basics.Online, basics.MicroAlgos{Raw: totalFunds})
+
+	bootstrappedNet.poolAddr = poolAddr
+	bootstrappedNet.sinkAddr = sinkAddr
+
+	//gensis block
+	initState, err := generateInitState(accounts, &bootstrappedNet)
+	if err != nil {
+		return err
+	}
+	localCfg := config.GetDefaultLocal()
+	localCfg.Archival = true
+	localCfg.CatchpointTracking = -1
+	localCfg.LedgerSynchronousMode = 0
+	log := logging.NewLogger()
+	l, err := ledger.OpenLedger(log, genesisFolder+"/bootstrapped", false, initState, localCfg)
+	if err != nil {
+		return err
+	}
+	l.WaitForCommit(0)
+
+	//create accounts, apps and assets
+	prev, _ := l.Block(l.Latest())
+	err = generateAccounts(src, fileCfgs.RoundTransactionsCount, prev, l, &bootstrappedNet, params)
+	if err != nil {
+		return err
+	}
+
+	//create more transactions
+	prev, _ = l.Block(l.Latest())
+	for i := uint64(bootstrappedNet.round); i < fileCfgs.NumRounds; i++ {
+		bootstrappedNet.round++
+		blk, _ := createBlock(src, prev, fileCfgs.RoundTransactionsCount, &bootstrappedNet, params)
+		blk.BlockHeader.TimeStamp += int64(crypto.RandUint64() % 100 * 1000)
+		err = l.AddBlock(blk, agreement.Certificate{Round: bootstrappedNet.round})
+		if err != nil {
+			fmt.Printf("Error %s\n", err)
+			return err
+		}
+		prev, _ = l.Block(l.Latest())
+	}
+
+	l.WaitForCommit(bootstrappedNet.round)
+	l.Close()
+
+	localCfg.CatchpointTracking = 0
+	l, err = ledger.OpenLedger(log, genesisFolder+"/bootstrapped", false, initState, localCfg)
+	if err != nil {
+		return err
+	}
+	l.Close()
+
+	fmt.Printf("ends %v\n", time.Now())
+	return nil
+}
+
+func getGenesisAlloc(name string, allocation []bookkeeping.GenesisAllocation) bookkeeping.GenesisAllocation {
+	for _, alloc := range allocation {
+		if strings.ToLower(alloc.Comment) == strings.ToLower(name) {
+			return alloc
+		}
+	}
+	return bookkeeping.GenesisAllocation{}
+}
+
+func keypair() *crypto.SignatureSecrets {
+	var seed crypto.Seed
+	crypto.RandBytes(seed[:])
+	s := crypto.GenerateSignatureSecrets(seed)
+	return s
+}
+
+func generateInitState(accounts map[basics.Address]basics.AccountData, bootstrappedNet *netState) (ledger.InitState, error) {
+
+	var initState ledger.InitState
+
+	block := bookkeeping.Block{
+		BlockHeader: bookkeeping.BlockHeader{
+			GenesisID:   bootstrappedNet.genesisID,
+			GenesisHash: bootstrappedNet.genesisHash,
+			Round:       bootstrappedNet.round,
+			RewardsState: bookkeeping.RewardsState{
+				RewardsRate: 1,
+				RewardsPool: bootstrappedNet.poolAddr,
+				FeeSink:     bootstrappedNet.sinkAddr,
+			},
+			UpgradeState: bookkeeping.UpgradeState{
+				CurrentProtocol: protocol.ConsensusCurrentVersion,
+			},
+			TxnCounter: 0,
+		},
+	}
+
+	initState.Block = block
+	initState.Accounts = accounts
+	initState.GenesisHash = crypto.Hash([]byte(bootstrappedNet.genesisID))
```

Comment:
Make sure you're consistent. The GenesisHash on the `initstate` need to match the one on the block header.
The one on the block header need to be calculated in the same way it was in `verifyGenesisHashes`.